### PR TITLE
New UI component - ImageStream

### PR DESCRIPTION
**Pull request by schlan**
##### Dialog
The old dialog is still there but marked as deprecated. 

screen shot 2017-01-29 at 20 49 33

#### Image stream

The new shiny UI component. 

belvedere

For displaying the last taken images the permission, `READ_EXTERNAL_STORAGE` is need on devices running API level 19 and above. 
The ImageStream adapts to the available permissions. 

- API level 18 and below: Most recent images are displayed
- API level 19 - 22: Most recent images will be shown If `READ_EXTERNAL_STORAGE` is in the manifest.
- API level 23 and above: Most recent images will be shown If `READ_EXTERNAL_STORAGE` is in the manifest and the user grants the permission. 
- API level 19 - 25: If `READ_EXTERNAL_STORAGE` is not it the manifest or the user denies the permission a list with available options will be displayed. (see screenshot)

screen shot 2017-01-29 at 21 07 17

### Reviewers
@brendan-fahy @baz8080 @schlan @eepDev

### Todo
- [x] Merge #24 
- [x] Merge #25

### CODE REVIEW

File: belvedere/src/main/res/values/styles.xml [hunk 1]
```diff
@@ -46,4 +46,25 @@
...
+    </style>
+
+
+    <style name="BottomSheetTheme" parent="Theme.AppCompat.Light.NoActionBar">
+        <item name="android:windowBackground">@android:color/transparent</item>
+        <item name="android:windowIsTranslucent">true</item>
+        <item name="android:windowNoTitle">true</item>
+        <item name="android:backgroundDimEnabled">true</item>
+
+        <item name="colorPrimary">#ffffff</item>
```
Comment (baz8080) [inline belvedere/src/main/res/values/styles.xml]: There should be a reliable `@android:color/white` if you want to use that

File: belvedere/src/main/res/menu/imagestream_menu.xml [hunk 2]
```diff
@@ -0,0 +1,7 @@
+<?xml version="1.0" encoding="utf-8"?>
+<menu xmlns:android="http://schemas.android.com/apk/res/android"
+      xmlns:app="http://schemas.android.com/apk/res-auto">
+    <item android:id="@+id/image_stream_system_gallery"
+          android:title="System gallery"
```
Comment (baz8080) [inline belvedere/src/main/res/menu/imagestream_menu.xml]: Extract to string resource

File: belvedere/src/main/java/zendesk/belvedere/ImageStreamModel.java [hunk 3]
```diff
@@ -0,0 +1,115 @@
...
+
+    @Override
+    public List<Uri> getLatestImages() {
+        final List<Uri> uris = new ArrayList<>();
+        final String[] projection = new String[]{
+                MediaStore.Images.ImageColumns._ID,
+                MediaStore.Images.ImageColumns.DATA
+        };
+
+        final Cursor cursor = context.getContentResolver()
```
Comment (baz8080) [inline belvedere/src/main/java/zendesk/belvedere/ImageStreamModel.java]: Don't you just <3 cursors!

Comment (schlan) [inline belvedere/src/main/java/zendesk/belvedere/ImageStreamModel.java]: ❤️

**Comment by schlan** [comment]
@baz8080 @brendan-fahy @eepDev read for review. Tests + docs will be added in a separate PR.

**Pull request merged**